taskcluster/ffios_taskgraph/transforms/bitrise.py

```diff
@@ -77,17 +77,29 @@ def set_worker_config(config, tasks):
 def add_bitrise_command(config, tasks):
     for task in tasks:
         commands = task["run"].setdefault("commands", [])
-        workflow = task.pop("bitrise-workflow")
-
-        command = [
-            "python3",
-            "taskcluster/scripts/bitrise-schedule.py",
-            "--token-file", ".bitrise_token",
-            "--branch", config.params["head_ref"],
-            "--commit", config.params["head_rev"],
-            "--workflow", workflow,
-            "--artifacts-directory", _ARTIFACTS_DIRECTORY
-        ]
+
+        if "bitrise-workflow" in task:
+            workflow = task.pop("bitrise-workflow")
+            command = [
+                "python3",
+                "taskcluster/scripts/bitrise-schedule.py",
+                "--token-file", ".bitrise_token",
+                "--branch", config.params["head_ref"],
+                "--commit", config.params["head_rev"],
+                "--workflow", workflow,
+                "--artifacts-directory", _ARTIFACTS_DIRECTORY
+            ]
+        else:
+            pipeline = task.pop("bitrise-pipeline")
+            command = [
+                "python3",
+                "taskcluster/scripts/bitrise-schedule.py",
+                "--token-file", ".bitrise_token",
+                "--branch", config.params["head_ref"],
+                "--commit", config.params["head_rev"],
+                "--pipeline", pipeline,
+                "--artifacts-directory", _ARTIFACTS_DIRECTORY
+            ]
```

Comment:
Having the ability to trigger pipelines seems like a nice feature to implement, and certainly doesn't hurt!

But I just wanted to mention that long term I think we'll want to have the pipeline logic be handled by Taskcluster where possible. I.e, try to have a 1 to 1 mapping of TC task <-> bitrise workflow. This way we benefit from the standard caching and optimization features we employ in our other projects.

To be clear there's no issue here, just flagging this as a potential thing we might want to work towards down the road.

Reply:
I see what you mean, there some pros and cons on doing that as how we have the pipelines integrated in the development workflow currently. While we monitor builds on Bitrise, if we split the pipeline, to be managed as separate workflows, when a build runs, it will be splitted also in the bitirse dashboard (for a pipeline we run 5 workflows, 4 of them in parallel). If there is a failure, for a developer or a test engineer, it may be difficult to find where the build is. Currently with the pipeline, a build, from building plus testing is just one entry where everything is shown together.
 
We can talk about this further and show you how it works now. I'm sure we can find a solution to this. Initially, as we start modifying things, I would suggest to use pipelines while we get more familiar with this and progressively change to worklfows and the new way of monitoring things on Bitrise / treeherder. 

So, completely agree on having this in mind for future potential changes.

Reply:
Ah ok, that makes sense then. I'm happy to keep it as is then. I really don't know much about bitrise, so I'm not surprised to learn there are tradeoffs I don't know about